clar2wasm/src/wasm_generator.rs

```diff
@@ -49,13 +49,15 @@ pub struct WasmGenerator {
 pub enum GeneratorError {
     NotImplemented,
     InternalError(String),
+    TypeError(String),
 }
 
 impl DiagnosableError for GeneratorError {
     fn message(&self) -> String {
         match self {
             GeneratorError::NotImplemented => "Not implemented".to_string(),
             GeneratorError::InternalError(msg) => format!("Internal error: {}", msg),
+            GeneratorError::TypeError(msg) => format!("Type error: {}", msg),
```

Comment:
Thanks for adding this. That is a cleaner way than just using the internal error. There are several other places where we can use this later too.